### PR TITLE
Fix dimension mismatch for ApplyBasisAtPoints

### LINKED ISSUE #1287: Bug in CeedBasisApplyAtPoints
Currently, the `basis->basis_chebyshev` created in `CeedBasisApplyAtPoints` is incorrectly created with `P=Q`. This is problematic for applying the basis with `CEED_NOTRANSPOSE`.

Specifically, right now we have:
```c
/** ceed-basis.c:1543 **/
    // Build basis mapping from nodes to Chebyshev coefficients
    CeedScalar       *chebyshev_interp_1d, *chebyshev_grad_1d, *chebyshev_q_weight_1d;
    const CeedScalar *interp_1d;

    CeedCall(CeedCalloc(Q_1d * Q_1d, &chebyshev_interp_1d));
    CeedCall(CeedCalloc(Q_1d * Q_1d, &chebyshev_grad_1d));
    CeedCall(CeedCalloc(Q_1d, &chebyshev_q_weight_1d));
    CeedCall(CeedBasisGetInterp1D(basis, &interp_1d));
    CeedCall(CeedMatrixMatrixMultiply(basis->ceed, chebyshev_coeffs_1d, interp_1d, chebyshev_interp_1d, Q_1d, P_1d, Q_1d));

    CeedCall(CeedVectorCreate(basis->ceed, num_comp * CeedIntPow(Q_1d, dim), &basis->vec_chebyshev));
    CeedCall(CeedBasisCreateTensorH1(basis->ceed, dim, num_comp, Q_1d, Q_1d, chebyshev_interp_1d, chebyshev_grad_1d, q_ref_1d, chebyshev_q_weight_1d,
                                     &basis->basis_chebyshev));
```
Here, `chebyshev_interp_1d` and `chebyshev_grad_1d` are both incorrectly created as square matrices (although the correct dimensions are used in the call to `CeedMatrixMatrixMultiply`). 

Instead, we should have the same dimensions for `basis->basis_chebyshev` as we do for `basis`:
```c
    // Build basis mapping from nodes to Chebyshev coefficients
    CeedScalar       *chebyshev_interp_1d, *chebyshev_grad_1d, *chebyshev_q_weight_1d;
    const CeedScalar *interp_1d;

    CeedCall(CeedCalloc(Q_1d * P_1d, &chebyshev_interp_1d));
    CeedCall(CeedCalloc(Q_1d * P_1d, &chebyshev_grad_1d));
    CeedCall(CeedCalloc(Q_1d, &chebyshev_q_weight_1d));
    CeedCall(CeedBasisGetInterp1D(basis, &interp_1d));
    CeedCall(CeedMatrixMatrixMultiply(basis->ceed, chebyshev_coeffs_1d, interp_1d, chebyshev_interp_1d, Q_1d, P_1d, Q_1d));

    CeedCall(CeedVectorCreate(basis->ceed, num_comp * CeedIntPow(Q_1d, dim), &basis->vec_chebyshev));
    CeedCall(CeedBasisCreateTensorH1(basis->ceed, dim, num_comp, P_1d, Q_1d, chebyshev_interp_1d, chebyshev_grad_1d, q_ref_1d, chebyshev_q_weight_1d,
                                     &basis->basis_chebyshev));
```

### PR DESCRIPTION
**Pull request by jeremylt**
Closes #1287 

Thanks for catching Zach. I tweaked a test to prevent regressions

### CODE REVIEW

File: interface/ceed-basis.c [hunk 1]
```diff
@@ -1551,7 +1551,7 @@ int CeedBasisApplyAtPoints(CeedBasis basis, CeedInt num_points, CeedTransposeMod
     CeedCall(CeedMatrixMatrixMultiply(basis->ceed, chebyshev_coeffs_1d, interp_1d, chebyshev_interp_1d, Q_1d, P_1d, Q_1d));
```
Comment (zatkins-dev) [inline interface/ceed-basis.c:1551]: GitHub won't let me comment on the actual lines (1547-1548), but shouldn't `chebyshev_interp_1d` and `chebyshev_grad_1d` both be size `Q_1d * P_1d`?

Comment (jeremylt) [reply interface/ceed-basis.c:1551]: I think I got it.

Comment (jeremylt) [reply interface/ceed-basis.c:1551]: I need to think if there is a way for memcheck or some bounds checking to catch this.

**Review by zatkins-dev — APPROVED** [review]
Looks good to me now!

**Pull request merged**